nix/modules/hardware.nix

```diff
@@ -15,19 +22,56 @@
       "xhci_pci"
       "ahci"
       "nvme"
+      "r8169"
+      "ixgbe"
+      "cdc_ether"
+      "virtio_net"
+      "virtio_pci"
+      "virtio_mmio"
+      "virtio_blk"
+      "virtio_scsi"
+      "9p"
+      "9pnet_virtio"
     ];
-    # / is a mirror raid
-    boot.loader.grub.devices = config.kuutamo.disko.disks;
+    # XXX on some platforms we pick up the wrong console. In this cases we default to serial,
+    # also our ipmi has VGA output, uncomment the line below.
+    srvos.boot.consoles = lib.mkDefault [ ];
 
     # Enable raid support specifically, this will disable srvos's
     # systemd-initrd as well, which currently is not compatible with mdraid.
     boot.swraid.enable = true;
     systemd.services.mdmonitor.enable = false;
 
-    # for mdraid 1.1
-    boot.loader.grub.extraConfig = "insmod mdraid1x";
     boot.loader.grub.enable = true;
     boot.loader.grub.efiSupport = true;
     boot.loader.grub.efiInstallAsRemovable = true;
+
+    # initrd networking for unlocking LUKS
+    boot.initrd.network = {
+      enable = true;
+      ssh = {
+        enable = true;
+        port = 2222;
+        authorizedKeys = config.users.extraUsers.root.openssh.authorizedKeys.keys;
+        hostKeys = [
+          "/var/lib/secrets/sshd_key"
+        ];
+      };
+      postCommands = ''
+        ip link set dev ${config.kuutamo.disko.networkInterface} up
```

Comment:
Are all these ip commands needed? `boot.initrd.network.enable = true` works fine for me.

Reply:
Not all providers do dhcp, so this is more generic.